package/MDAnalysis/analysis/hbonds/hbond_analysis.py

```diff
@@ -594,9 +595,13 @@ def __init__(self, universe, selection1='protein', selection2='all', selection1_
         self.distance = distance
         self.distance_type = distance_type  # note: everything except 'heavy' will give the default behavior
         self.angle = angle
-        self.traj_slice = slice(start if isinstance(start, int) else None,  # internal frames are 0 based
-                                stop if isinstance(stop, int) else None,
-                                step)
+        slice_index = []
+        for index in [start, stop, step]:
+            if isinstance(index, numbers.Integral):
+                slice_index.append(int(index))
+            else:
+                slice_index.append(None)
```

Comment:
@xiki-tempula Yes, this would be the best solution. Then any invalid value would just trigger an exception.